### ISSUE TITLE
Release Minor DataFusion 43.1.0 release

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#13499]
### Is your feature request related to a problem or challenge?

- We released https://github.com/apache/datafusion/issues/12470
- Delta rs is still having trouble updating: https://github.com/delta-io/delta-rs/pull/2886

@timsaucer mentioned  he has have found several items while working on delta table integration and would like to get a small incremental release to fix them

### List of items to include
- [x] https://github.com/apache/datafusion/issues/13481

**Comment by alamb** [comment]
I am happy to run the release process to make this release

**Comment by timsaucer** [comment]
Thank you. Right now https://github.com/apache/datafusion/issues/13481 is a blocker for delta-ra

**Comment by gruuya** [comment]
I thought https://github.com/apache/datafusion/pull/13468 resolves https://github.com/apache/datafusion/issues/13481 (again only keeping the second assert in `test_simplify_case_cast_list`)? If so can we proceed with merging it?

**Comment by rtyler** [comment]
Thanks for pushing forward here @timsaucer ! I believe that I have resolved the last of our issues in delta-rs but in order to avoid problems we have needed to disable some default optimizers. The changes you're working on would allow us to resume usage of the default optimizers. 

Either way, I'm hoping to land our pull request and cut a release soon either way :smiling_imp:

**Comment by timsaucer** [comment]
Ok, we do still have a handful of uft8view issues to address in https://github.com/apache/datafusion/issues/13504 but I think we've gotten all the blockers for delta-rs and datafusion-python once #13481 goes in. Then I think it would make sense to go ahead with this minor release and we'll get all the rest of the remaining utf8view issues in DF44.

**Comment by timsaucer** [comment]
@alamb is there anything I can do to help with this?

**Comment by alamb** [comment]
Hi @timsaucer  -- if you could create PRs with the necessary changes that target the `branch-43` branch that would be awesome: https://github.com/apache/datafusion/tree/branch-43

The idea is that I will then create a release candidate from that branch

This is described here: https://github.com/apache/datafusion/blob/18fc103a403ab0efe5245dd4352f3f3b93c2a4fe/dev/release/README.md#branching-policy

In the past, what I have done is something like
```shell
git checkout branch-43
git checkout -b backport_to_43
git cherry-pick <sha_from_master>
git push -u alamb
```

And then create a PR that targets `branch-43` rather than `main`

Does that make sense?

**Comment by timsaucer** [comment]
Ok, I can do that. In the past did you cherry pick everything from the point of the past release up to current in main? Or did you only grab specific commits you wanted?

**Comment by alamb** [comment]
> Ok, I can do that. In the past did you cherry pick everything from the point of the past release up to current in main? Or did you only grab specific commits you wanted?

Normally I would suggest we only cherry-pick the commits we want.

**Comment by alamb** [comment]
Is anyone waiting on this release? If not let's focus our efforts on
- https://github.com/apache/datafusion/issues/13334

And ideally 
- https://github.com/apache/datafusion/issues/13661

**Comment by gruuya** [comment]
> Is anyone waiting on this release?

I don't think so; delta-rs has a temp work-around for #13481, and 44 is just around the corner so we might as well go for that instead.

**Comment by alamb** [comment]
I agree -- it would be great to get as much effort as possible into making 44 a stable, easy to use release

**Comment by alamb** [comment]
Closing for now, please comment / let me know if anyone disagrees
